olp-cpp-sdk-dataservice-read/src/repositories/PrefetchTilesRepository.cpp

```diff
@@ -169,6 +171,30 @@ SubQuadsResponse PrefetchTilesRepository::GetSubQuads(
   OLP_SDK_LOG_TRACE_F(kLogTag, "GetSubQuads(%s, %" PRId64 ", %" PRId32 ")",
                       tile.ToHereTile().c_str(), version, depth);
 
+  repository::PartitionsCacheRepository repository(
+      catalog, settings.cache, settings.default_cache_expiration);
+
+  auto get_sub_quads = [](const QuadTreeIndex& tree) -> SubQuadsResult {
+    SubQuadsResult result;
+    for (const auto& data : tree) {
+      if (data) {
+        result.emplace(data->tile_key, data->data_handle);
+      }
+    }
```

Comment:
I would expect this to be placed in QuadTreeIndex. Why do you need to expose the iterators when we can place all these helper methods inside QuadTreeIndex?